project-koku/koku
# Suspend provider operations while in misconfigured state

A provider will be considered active when the boolean flag `active` is True on the provider table.  For this PR the `active` flag is marked false when a provider fails the cloud account checks via the `PUT /provider` method.

`active` flag integration can be included in a future PR once the sources status endpoint is in place (https://github.com/project-koku/koku/issues/1309 ).  When the check fails `active = false` when it's successful `active = true`.

Platform Sources is providing an API for applications to push detailed error strings to them.  The Cost management UI can know when a provider is having problems by monitoring this flag and then could provide a link to the Platform Sources UI to get the detailed status (UX mocks underway..)

While a provider is marked as not-active, it will no longer be ingested until the active status is restored.

**Testing**
1.  Add a provider with bad values.  Verify it's not created
2.  Add a provider with good values.  Verify it's created and active flag is set in database
3.  PUT /provider to change the bucket name and verify that the active flag is false.  Attempt to ingest data and verify that download/processing is blocked
4. Fix the provider by adding PUT /provider with the correct bucket name.  Use /download endpoint to trigger download/process.  Verify it's successful.

**Test Results**
[provider_pause_ut.txt](https://github.com/project-koku/koku/files/3806373/provider_pause_ut.txt)



Addresses #1326
